ld/bootloader.ld

```diff
@@ -99,7 +105,9 @@ SECTIONS {
 	.unmapped 0 : { *(*) }
 }
 
+
 ASSERT(SIZEOF(.unmapped) == 0, "SIZEOF(.unmapped) != 0")
+ASSERT(SIZEOF(.app) <= LENGTH(APPFLASH), "SIZEOF(.app) > LENGTH(APPFLASH)")
```

Comment:
```suggestion
ASSERT(SIZEOF(.app) > 0, "SIZEOF(.app) == 0")
```
The linking process will fail if `.app` exceeds `APPFLASH` either way. It's better to check if there even is an app since you're expecting one.